### ISSUE TITLE
Watchlist not syncing

### ISSUE THREAD

**Issue by EnSt0rm** [Taxel/PlexTraktSync#1940]
### Confirmation

- [X] I have read the [README.md](https://github.com/Taxel/PlexTraktSync#readme) on the project homepage
- [X] I have checked if identical issue already exists
- [ ] I have tried downgrading to find version that can be used as a workaround

### The problem

I run the sync command, and my collection and watched status updates properly, but my Plex Watchlist will not sync across to my Trakt Watchlist or vice-versa.
I've checked my config file and `plex_to_trakt` has `watchlist: true`, and `trakt_to_plex` also has `watchlist: true`.
I've tried `watchlist_as_playlist: false` and `true`, doesn't seem to make any difference.

### Steps to reproduce the behavior

docker-compose run --rm plextraktsync sync

### Expected behavior

Watchlist to be synced between Plex universal watchlist, and Trakt watchlist

### Config file contents

```yaml
cache:
  path: /app/config/trakt_cache
excluded-libraries:
- Private
- Family Holidays
config:
  dotenv_override: true
plex:
  timeout: 30
logging:
  append: true
  console_time: false
  debug: false
  filename: plextraktsync.log
  filter_loggers: null
  filter: null
sync:
  rating_priority: plex
  plex_to_trakt:
    collection: true
    clear_collected: false
    ratings: true
    watched_status: true
    watchlist: true
  trakt_to_plex:
    liked_lists: true
    ratings: true
    watched_status: true
    watchlist: true
    watchlist_as_playlist: false
    playback_status: false
watch:
  add_collection: false
  remove_collection: false
  scrobble_threshold: 80
  username_filter: true
  media_progressbar: true
  ignore_clients: null
xbmc-providers:
  movies: imdb
  shows: tvdb
```

### Install method

docker-compose

### Version

0.30.4

### Python Version

3.12.3

### Plex Server Version

1.40.2.8395

### Operating System and Version

Linux-5.15.146.1-microsoft-standard-WSL2-x86_64-with

**Comment by glensc** [comment]
> Workarounds
>
> No response

Downgrade to what version fixes problem? also share sync log, i.e. what plugins get enabled.

**Comment by EnSt0rm** [comment]
Haven't tried downgrading version yet, I'll try some downgrade versions shortly to see if that makes a difference.

In the meantime, here's the sync log from attempting to run this morning:

2024-05-11 10:04:57,125 INFO[plextraktsync.commands.sync]:PlexTraktSync [0.30.5]
2024-05-11 10:04:57,588 INFO[plextraktsync.plex.PlexServerConnection]:Connecting with url: https://************:32400, timeout 30 seconds
2024-05-11 10:04:58,003 INFO[plextraktsync.commands.sync]:Sync Movie sections: ['3D Movies', 'Movies', 'Movies - 4K', 'Movies - Kids']
2024-05-11 10:04:58,018 INFO[plextraktsync.commands.sync]:Sync Show sections: ['TV Shows', 'TV Shows - 4K', 'TV Shows - Kids']
2024-05-11 10:04:58,066 INFO[plextraktsync.sync.plugin.SyncPluginManager]:Enable sync plugin 'AddCollectionPlugin': True
2024-05-11 10:04:58,073 INFO[plextraktsync.sync.plugin.SyncPluginManager]:Enable sync plugin 'ClearCollectedPlugin': False
2024-05-11 10:04:58,076 INFO[plextraktsync.sync.plugin.SyncPluginManager]:Enable sync plugin 'LikedListsPlugin': True
2024-05-11 10:04:58,085 INFO[plextraktsync.sync.plugin.SyncPluginManager]:Enable sync plugin 'SyncRatingsPlugin': True
2024-05-11 10:04:58,088 INFO[plextraktsync.sync.plugin.SyncPluginManager]:Enable sync plugin 'SyncWatchedPlugin': True
2024-05-11 10:04:58,094 INFO[plextraktsync.sync.plugin.SyncPluginManager]:Enable sync plugin 'TraktListsPlugin': True
2024-05-11 10:04:58,099 INFO[plextraktsync.sync.plugin.SyncPluginManager]:Enable sync plugin 'WatchListPlugin': True
2024-05-11 10:05:05,903 INFO[plextraktsync.sync.plugin.SyncPluginManager]:Enable sync plugin 'WatchProgressPlugin': False
2024-05-11 10:05:06,908 WARNING[plextraktsync.sync.TraktListsPlugin]:Disabling TraktListsPlugin: No lists to process
2024-05-11 10:05:14,807 INFO[plextraktsync.decorators.measure_time]:3D Movies processed in 7.89 seconds
2024-05-11 10:12:06,664 INFO[plextraktsync.decorators.measure_time]:Movies processed in 6 minutes and 51.86 seconds
2024-05-11 10:12:08,086 INFO[plextraktsync.decorators.measure_time]:Movies - 4K processed in 1.42 seconds
2024-05-11 10:13:20,669 INFO[plextraktsync.decorators.measure_time]:Movies - Kids processed in 1 minute and 12.58 seconds
2024-05-11 10:13:20,672 INFO[plextraktsync.plan.Walker]:Preload shows data
2024-05-11 10:13:25,279 INFO[plextraktsync.decorators.measure_time]:TV Shows processed in 4.61 seconds
2024-05-11 10:13:25,378 INFO[plextraktsync.decorators.measure_time]:TV Shows - 4K processed in 86 milliseconds and 519 microseconds
2024-05-11 10:13:26,565 INFO[plextraktsync.decorators.measure_time]:TV Shows - Kids processed in 1.18 seconds
2024-05-11 10:13:26,567 INFO[plextraktsync.plan.Walker]:Preloaded shows data (889 shows)
2024-05-11 10:13:26,572 INFO[plextraktsync.plan.Walker]:Preload shows matches
2024-05-11 10:17:37,419 INFO[plextraktsync.plan.Walker]:Preloaded shows matches (889 shows)
2024-05-11 10:28:35,033 INFO[plextraktsync.decorators.measure_time]:TV Shows processed in 10 minutes and 57.63 seconds
2024-05-11 10:28:36,489 INFO[plextraktsync.decorators.measure_time]:TV Shows - 4K processed in 1.43 seconds
2024-05-11 10:44:41,196 INFO[plextraktsync.decorators.measure_time]:TV Shows - Kids processed in 16 minutes and 4.71 seconds
2024-05-11 10:44:41,958 INFO[plextraktsync.decorators.measure_time]:Completed full sync in 39 minutes and 44.06 seconds

**Comment by EnSt0rm** [comment]
Found a downgrade version that sync is correctly working with watchlists, 0.29.14 works, 0.30.0 through to latest don't work.

**Comment by glensc** [comment]
Pretty nasty bug when refactored. Oneliner to fix it: https://github.com/Taxel/PlexTraktSync/pull/1942

**Comment by HomerBeer** [comment]
> Pretty nasty bug when refactored. Oneliner to fix it: #1942

**Comment by HomerBeer** [comment]
Got the same issue, how did above fix work?
